### ISSUE TITLE
VM IP static configuration

### ISSUE THREAD

**Issue by joppino** [oVirt/go-ovirt-client#241]
Hello, I've opened a similar issue in the terraform ovirt provider which is using this client.

It seems no more possibile to configure the VM main NIC using the initialization parameter nic_configuration as before with the terraform provider.

```
 initialization {
          dns_search  = "example.com"
          dns_servers = "8.8.8.8"
          host_name   = "temp01.example.com"
          timezone    = "Africa/Nairobi"

          nic_configuration {
              address    = "192.168.10.11"
              boot_proto = "static"
              gateway    = "192.168.10.254"
              label      = "eth0"
              netmask    = "255.255.255.0"
              on_boot    = true
            }
        }
```
It seems that this is due to the fact that in the current ovirt-client version  (v2.1.0) the Initialization contains a Cloud-init custom script and/or the initialization hostname, and nothing more:

```
// Initialization defines to the virtual machine’s initialization configuration.
type Initialization interface {
	CustomScript() string
	HostName() string
}
```

So the question is how to replicate the previous behavior. The Ansible module still lets the user specify the initial nic_configuration, but I cannot find a way to replicate this in the current version. 

Thanks

**Comment by engelmi** [comment]
Hi @joppino, you are right - this feature is currently missing in the `go-ovirt-client` and we would need to add it here in order to enable this in the [oVirt terraform provider](https://github.com/oVirt/terraform-provider-ovirt). 
The `Initialization` interface should be the right place to extend. I'll have a look at it.

**Comment by dyens** [comment]
Hi!
@engelmi I would like to add all `Initialization` arguments to `go-ovirt-client`. I think, that all of them can be very useful.
I found, that you already work on `NicConfiguration` parameter https://github.com/oVirt/go-ovirt-client/commit/cf304dbb00d72f3345e6e674ee9b88299d8c6375. 

After merge your feature, i would like to create PR for other params (and pr for terraform provider). What do you think ?

Also i think, that all of `Initialization` parameters are optional. 
What do you think if save `customScript` and `hostname` in constructor for backward compatibility:

```go
func NewInitialization(customScript, hostname string) BuildableInitialization {..} 
```

And all other parameters should be configured via builder:

```go

initialization := NewInitialization("custom-script", "hostname").WithSomeParam("some-param");
```

**Comment by engelmi** [comment]
Hi @dyens, I created an issue for this (#250) - feel free to assign it to you. I have limited capacity, but I'll make sure to review your contribution and answer questions if some arise.

**Comment by engelmi** [comment]
@joppino This has been implemented in #249. I'll create a new release soon. Closing this issue.

**Comment by dyens** [comment]
@engelmi `TestVMCreationWithInit` in live tests mode does not pass. This PR should fix: https://github.com/oVirt/go-ovirt-client/pull/251
